cocos/2d/components/sprite.ts

```diff
@@ -696,6 +692,10 @@ export class Sprite extends UIRenderer {
             }
             if (textureChanged) {
                 if (this.renderData) this.renderData.textureDirty = true;
+                // texture type changed, set this._instanceMaterialType to default value
+                if ((oldFrame ? oldFrame.texture instanceof RenderTexture : false) !== spriteFrame.texture instanceof RenderTexture) {
```

Comment:
When judging texturechange earlier, we have ensured that oldFrame.texture is not equal to spriteFrame.texture？

Reply:
So what I'm judging here is whether the texture is changing from RenderTexture to a regular Texture, or from a regular Texture to RenderTexture.

Reply:
The code looks wired. Could you please modify it to make it more clear?

Reply:
ok, got it